code/game/objects/items/AI_modules/hacked.dm

```diff
@@ -34,3 +34,49 @@
 			law_datum.replace_random_law(laws[1], list(LAW_ION, LAW_HACKED, LAW_INHERENT, LAW_SUPPLIED), LAW_HACKED)
 	return laws[1]
 
+/obj/item/ai_module/malf // Gives syndie laws as well as making AI malf
+	name = "Infected AI Module"
+	desc = "An AI Module, infected with a virus."
+	bypass_law_amt_check = TRUE
+	laws = list("")
+	var/functional = TRUE
+
+/obj/item/ai_module/malf/transmitInstructions(datum/ai_laws/law_datum, mob/sender, overflow)
+	if(!sender.mind?.has_antag_datum(/datum/antagonist/traitor))
+		to_chat(sender, span_warning("You have no clue how to use this thing."))
+		return
+	if(!functional)
+		to_chat(sender, span_warning("It is broken and non-functional, what do you want from it?"))
+	var/mob/living/silicon/ai/malf_candidate = law_datum.owner
+	if(!istype(malf_candidate)) //If you are using it on cyborg upload console or a cyborg
+		to_chat(sender, span_warning("You should use [src] on an AI upload consoles or the AI core itself."))
+		return
+	if(malf_candidate.mind?.has_antag_datum(/datum/antagonist/malf_ai)) //Already malf
+		to_chat(sender, span_warning("Unknown error occured. Upload process aborted."))
+		return
+	malf_candidate.laws = new /datum/ai_laws/syndicate_override
+	var/datum/antagonist/malf_ai/malf_datum = new (give_objectives = FALSE)
+	malf_datum.name = "Infected AI"
+	malf_datum.employer = "Infected AI"
+	malf_datum.give_zeroth_laws = FALSE
+	malf_candidate.mind.add_antag_datum(malf_datum)
+	malf_candidate.set_zeroth_law("Only [sender.real_name] and people [sender.p_they()] designate[sender.p_s()] as being such are Syndicate Agents.")
+	var/datum/objective/protect/protection_objective = new
```

Comment:
```suggestion
	malf_candidate.laws = new /datum/ai_laws/syndicate_override

	var/datum/antagonist/malf_ai/malf_datum = new (give_objectives = FALSE)
	malf_datum.name = "Infected AI"
	malf_datum.employer = "Infected AI"
	malf_datum.give_zeroth_laws = FALSE
	malf_candidate.mind.add_antag_datum(malf_datum)
	malf_candidate.set_zeroth_law("Only [sender.real_name] and people [sender.p_they()] designate[sender.p_s()] as being such are Syndicate Agents.")

	var/datum/objective/protect/protection_objective = new
```

Space it out, quite hard to read since it all runs together. Brownie points would be to spin some of this stuff it out into it's own proc that we invoke in this proc for the best visibility in the world, like

```txt
/// Returns a datum that we can apply to our AI.
/obj/item/ai_module/malf/proc/generate_datum
    ...
```

Then, you could do something like:
```txt
malf_candidate.mind.add_antag_datum(generate_datum)
```

In this proc. However you want to split it up is up to you, but I think it might be a bit more clear rather than have one huge proc that we do several things in... it causes a lot of the variable names to mush together (i myself had my head spin a bit trying to keep track of malf_datum versus malf_candidate).